### ISSUE TITLE
[BUG/ISSUE] Wrong units in function GET_OH in sulfate_mod.F

### ISSUE THREAD

**Issue by lizziel** [geoschem/geos-chem#166]
See next comment for full issue description.

**Comment by lizziel** [comment]
Issue reported by Rong Tian (NUIST):

> When I read the code in sulfate_mod(Version 12.6.0), I am a little confused about the unit conversion in Function GET_OH.
>  
> In GeosCore/sulfate_mod.F, Line 7379 (FUNCTION GET_OH):
>  
> ```
>                   ! Take OH from State_Chm%Species [molec/cm3]
>                   ! OH is defined only in the chemistry grid
>                   IF ( State_Met%InChemGrid(I,J,L) ) THEN
>                      OH_MOLEC_CM3 = State_Chm%Species(I,J,L,id_OH)
>                  ELSE
>                      OH_MOLEC_CM3 = 0e+0_fp
>                  ENDIF
> ```
>  
> However, I have checked the code and printed out some variables (e.g. State_Chm%Spc_Units) and found that State_Chm%Species here are in the unit of v/v dry, not molec/cm3. Also, the Function GET_OH in sulfate_mod is different from GET_OH in carbon_mod. The latter one includes the unit conversion from kg to molec/cm3. So I wondered if the GET_OH function in sulfate_mod should be modified to convert the unit from v/v dry to molec/cm3.
>  
> I'm not sure what I understand is right or not. So I'm writing to ask you team for help and advice. I'm looking forward to receiving your answer at your first convenience!
>  
> Thanks much!
>
> 
>  
>  
> Best,
>  
>  
> Rong Tian

**Comment by lizziel** [comment]
I believe the missing conversion from v/v to molec/cm3 in GET_OH in sulfate_mod.F is indeed a bug. The analogous GET_OH conversion in carbon_mod.F that you point out was added several years ago as a bug fix. The sulfate_mod.F GET_OH bug impacts the following:

- Aerosol-only simulation:
  - H2O2 calculation in CHEM_H2O2
  - Gas phase SO4 production rate calculation in CHEM_SO2 (offline only)
- Acid uptake full-chemistry simulation:
  - H2SO4 calculation in CHEM_SO2

We can add this bug fix to 12.7. Thanks for bringing this to our attention.

**Comment by lizziel** [comment]
Actually, looking the at the subroutine GET_OH more closely, apart from where it is called, the bug will NOT impact aerosol-only simulations. The code snippet below show that OH is calculated in a different way for aerosol-only simulations:
https://github.com/geoschem/geos-chem/blob/5f25c0fc6ff27ec076d766ac178554d9b91bf636/GeosCore/sulfate_mod.F#L7413

This means the bug impacts only the acid update full-chemistry simulation, or custom simulations in which users turn on acid uptake on dust.

**Comment by lizziel** [comment]
See below for the bug fix which will go into 12.7 after testing:
```
            ! Get OH from State_Chm%Species [v/v] converted to [molec/cm3]                                         
            MolecRatio = State_Chm%SpcData(id_OH)%Info%MolecRatio
            MW_g = State_Chm%SpcData(id_OH)%Info%emMW_g
            OH_MOLEC_CM3 = State_Chm%Species(I,J,L,id_OH)
     &                     * State_Met%AIRDEN(I,J,L) * 1e+3_fp * AVO
     &                     / ( AIRMW * MolecRatio )
```

**Comment by yantosca** [comment]
@lizziel added this fix as commit: https://github.com/geoschem/geos-chem/commit/03c0fff792d009db7ca9cbe35503a5fca4cc36a6
